pyro/contrib/minipyro.py

```diff
@@ -78,6 +92,19 @@ def process_message(self, msg):
             msg["value"] = self.guide_trace[msg["name"]]["value"]
 
 
+# block allows the selective application of effect handlers to different parts of a model.
+# Sites hidden by block will only have the handlers below block on the PYRO_STACK applied,
+# allowing inference or other effectful computations to be nested inside models.
+class block(Messenger):
+    def __init__(self, fn, hide_fn):
```

Comment:
How about defaulting `fn=None, hide_fn=lambda msg: True`? Then we can use the syntax
```py
with block():
    ...do stuff...
```
Plus it doesn't come out of our 500-line budget :wink: